### PR TITLE
Use existing helm template logic to populate manifests instead of dry-run

### LINKED ISSUE #1712: Can't deploy kube-prometheus-stack with helm.v3.Release
Can't deploy kube-prometheus-stack with helm.v3.Release.  It appears that pulumi is trying to build kubernetes objects for the CRDs that are not yet installed.

## Steps to reproduce

```ts
const prometheusNamespace = new k8s.core.v1.Namespace("prometheus", {
    metadata: {
        name: "prometheus",
    },
});
const prometheusRelease = new k8s.helm.v3.Release("prometheus", {
    name: "kube-prometheus-stack",
    chart: "kube-prometheus-stack",
    repositoryOpts: {
        repo: "https://prometheus-community.github.io/helm-charts",
    },
    version: "18.0.8",
    namespace: prometheusNamespace.metadata.name,
    values: {},
}, { dependsOn: [prometheusNamespace], provider: cluster.provider });
```

Expected: Deployment of kube-prometheus-stack with CRDs
Actual: Fails with error as the CRDs the chart installs are not already installed.

```
Diagnostics:
  kubernetes:helm.sh/v3:Release (prometheus):
    error: unable to build kubernetes objects from release manifest: [unable to recognize "": no matches for kind "Alertmanager" in version "monitoring.coreos.com/v1", unable to recognize "": no matches for kind "Prometheus" in version "monitoring.coreos.com/v1", unable to recognize "": no matches for kind "PrometheusRule" in version "monitoring.coreos.com/v1", unable to recognize "": no matches for kind "ServiceMonitor" in version "monitoring.coreos.com/v1"]
```

### PR DESCRIPTION
**Pull request by viveklak**
Fixes #1712 

This avoids issues for charts with CRDs which don't play well with dry-run.

### CODE REVIEW

File: tests/sdk/nodejs/examples/helm-release-crd/step2/index.ts [hunk 1]
```diff
@@ -0,0 +1,15 @@
+import * as k8s from "@pulumi/kubernetes";
+
+const prometheusRelease = new k8s.helm.v3.Release("prometheus", {
+    name: "kube-prometheus-stack",
+    chart: "kube-prometheus-stack",
+    repositoryOpts: {
+        repo: "https://prometheus-community.github.io/helm-charts",
+    },
+    values: {
+        commonLabels: {
```
Comment (mikhailshilkov) [inline tests/sdk/nodejs/examples/helm-release-crd/step2/index.ts:10]: Nit: add a comment somewhere about the purpose of this test (what step2 is verifying)

**Review by mikhailshilkov — APPROVED** [review]
This makes sense to me, would love @lblackstone to take a look too

File: tests/sdk/nodejs/examples/helm-release-crd/step2/index.ts (hunk 1 above)
Comment (viveklak) [reply tests/sdk/nodejs/examples/helm-release-crd/step2/index.ts:10]: Done

**Review by lblackstone — COMMENTED** [review]
This worked when I tested it locally, but I had a couple questions:

1. Are we planning to render the manifests in a separate PR? We should have all the info we need to do that now similarly to our Chart support.
2. I got a number of warnings during the update. Any idea where those are coming from?
```
    E0917 10:12:09.637652   63178 memcache.go:196] couldn't get resource list for monitoring.coreos.com/v1alpha1: the server could not find the requested resource
    E0917 10:12:09.748362   63178 memcache.go:196] couldn't get resource list for monitoring.coreos.com/v1: the server could not find the requested resource
    E0917 10:12:09.777696   63178 memcache.go:196] couldn't get resource list for monitoring.coreos.com/v1: the server could not find the requested resource
    E0917 10:12:09.839002   63178 memcache.go:196] couldn't get resource list for monitoring.coreos.com/v1: the server could not find the requested resource
```

**Comment by viveklak** [comment]
> This worked when I tested it locally, but I had a couple questions:
> 
> 1. Are we planning to render the manifests in a separate PR? We should have all the info we need to do that now similarly to our Chart support.

This is tracked in https://github.com/pulumi/pulumi-kubernetes/issues/1701. We have had that support with dry-run as well but maps were not rendering. I can try rendering as formatted yaml instead but that would be in the context of a separate PR.

File: tests/sdk/nodejs/examples/helm-release-crd/step1/index.ts [hunk 2]
```diff
@@ -0,0 +1,11 @@
+import * as k8s from "@pulumi/kubernetes";
+
+const prometheusRelease = new k8s.helm.v3.Release("prometheus", {
+    name: "kube-prometheus-stack",
+    chart: "kube-prometheus-stack",
+    repositoryOpts: {
```
[the remark this answers is not in the document]
Comment (viveklak) [reply tests/sdk/nodejs/examples/helm-release-crd/step1/index.ts:7]: Pinned version.

**Comment by viveklak** [comment]
> 2. I got a number of warnings during the update. Any idea where those are coming from?
> 
> ```
>     E0917 10:12:09.637652   63178 memcache.go:196] couldn't get resource list for monitoring.coreos.com/v1alpha1: the server could not find the requested resource
>     E0917 10:12:09.748362   63178 memcache.go:196] couldn't get resource list for monitoring.coreos.com/v1: the server could not find the requested resource
>     E0917 10:12:09.777696   63178 memcache.go:196] couldn't get resource list for monitoring.coreos.com/v1: the server could not find the requested resource
>     E0917 10:12:09.839002   63178 memcache.go:196] couldn't get resource list for monitoring.coreos.com/v1: the server could not find the requested resource
> ```

Thanks for pointing that out. I saw them as well. Looks like these were spurious log messages from the stock discovery cache client around CRD installs (perhaps defaulting to stdout). Switched to version in clients and the log messages are gone.

**Review by lblackstone — APPROVED** [review]

**Comment by viveklak** [comment]
There seems to be something going on with the CI run - Levi and I have independently verified the new test locally and I have run it on an identical GKE cluster to the CI environment. All of them work but it seems the released provider is being somehow pulled into the test run on GHA. I am going to force merge this and watch the master build to see what is up.

**Pull request merged**